### ISSUE TITLE
Unify fragmented compression-metric prediction into clio_ctp/compress/model

### ISSUE THREAD

**Issue by lukemartinlogan** [iowarp/core#693]
## Description

Compression-metric prediction code (models that predict compression ratio /
throughput from data statistics) is currently **fragmented and duplicated**
across several repositories and directories:

- `context-transport-primitives/src/compression/` — C++ predictors
  (`xgboost_predictor.cc`, `qtable_predictor.cc`, `dense_nn_predictor.cc`,
  `linreg_table_predictor.cc`)
- `context-transport-primitives/include/clio_ctp/compress/data_stats.h` —
  data statistics / feature extraction
- `context-transfer-engine/compressor/` — benchmarks, a synthetic training-data
  generator, CSVs, and Python training scripts
  (`train_all_models.py`, `train_optimal_qtable.py`, `dnn_hyperparam_search.py`,
  `qtable_hyperparam_study.py`)
- The external **NeuroPress** repo (`git@github.com:grc-iit/NeuroPress.git`) —
  a dedicated NN compression-metric model, preprocessors, and a GPU/CPU
  synthetic compressor benchmark

This duplication makes the models hard to maintain, retrain, and reuse.

## Goal

Create a single, unified compression-metric modeling module under
`context-transport-primitives` that consolidates all of the above. NeuroPress is
cloned under `external/NeuroPress` **for reference (NOT a git submodule)** and
its reusable parts are ported in.

New layout:

```
context-transport-primitives/
  include/clio_ctp/compress/preprocess/   # feature extraction / preprocessors
  include/clio_ctp/compress/model/        # C++ stat-model headers
  src/compress/model/                      # C++ stat-model impls
  test/unit/compress/model/
  test/unit/compress/preprocess/
```

## Scope / Deliverables

1. **Synthetic compressor benchmark → CSV.** Combine the NeuroPress benchmark and
   `context-transfer-engine/compressor` benchmark into one tool that runs CPU
   **and** GPU compressors (GPU optional / compiled-out when unavailable) over
   synthetic (and optionally real) data and emits a training CSV.
2. **Preprocessors** ported from NeuroPress into
   `include/clio_ctp/compress/preprocess` (unified with the existing
   `data_stats.h`).
3. **Training scripts** for xgboost, qtable, and the NeuroPress neural net
   (the NeuroPress NN specifically, *not* a plain dense NN).
4. **C++ compression stat models** for xgboost, qtable, and the NeuroPress NN,
   under `model/`, sharing one predictor interface.

## Acceptance Criteria

- [ ] New directory layout created and wired into CMake (GPU-optional guarded)
- [ ] Unified predictor interface; xgboost / qtable / NN models implement it
- [ ] Preprocessors consolidated under `compress/preprocess`
- [ ] Synthetic benchmark builds CPU-only and (when CUDA available) with GPU
      compressors, producing a documented CSV schema
- [ ] Training scripts consume that CSV schema and emit loadable model artifacts
- [ ] Unit tests under `test/unit/compress/{model,preprocess}` (simple_test.h)
- [ ] No duplicated predictor code left behind (old paths migrated or shimmed)
